crates/bevy_color/src/color_ops.rs

```diff
@@ -0,0 +1,43 @@
+/// Methods for changing the luminance of a color. Note that these methods are not
+/// guaranteed to produce the same results in different color spaces, but are only
+/// approximations.
+pub trait Luminance: Sized {
+    /// Return the luminance of this color (0.0 - 1.0).
+    fn luminance(&self) -> f32;
+
+    /// Return a new version of this color with the given luminance.
+    fn with_luminance(&self, alpha: f32) -> Self;
+
+    /// Return a darker version of this color. The `amount` should be between 0.0 and 1.0.
+    /// The amount represents an absolute decrease in luminance, and is commutative:
+    /// `color.darker(a).darker(b) == color.darker(a + b)`.
+    ///
+    /// For a relative decrease in luminance, you can simply `mix()` with black.
+    fn darker(&self, amount: f32) -> Self;
+
+    /// Return a lighter version of this color. The `amount` should be between 0.0 and 1.0.
+    /// The amount represents an absolute increase in luminance, and is commutative:
+    /// `color.lighter(a).lighter(b) == color.lighter(a + b)`.
+    ///
+    /// For a relative increase in luminance, you can simply `mix()` with white.
+    fn lighter(&self, amount: f32) -> Self;
+}
+
+/// Linear interpolation of two colors within a given color space.
+pub trait Mix: Sized {
+    /// Linearly interpolate between this and another color, by factor.
+    /// Factor should be between 0.0 and 1.0.
+    fn mix(&self, other: &Self, factor: f32) -> Self;
+
+    /// Linearly interpolate between this and another color, by factor, storing the result
+    /// in this color. Factor should be between 0.0 and 1.0.
+    fn mix_assign(&mut self, other: Self, factor: f32) {
+        *self = self.mix(&other, factor);
+    }
+}
+
+/// Methods for manipulating alpha values.
+pub trait Alpha: Sized {
```

Comment:
This should probably have a `alpha` method to get the alpha as well.

Reply:
It's very useful when working with trait objects.

Reply:
Done. (Will be in next push).